### ISSUE TITLE
Traceback (most recent call last):

### ISSUE THREAD

**Issue by lily980905** [jaysonlong/webvideo-downloader#15]
Traceback (most recent call last):
  File "C:\Users\lily\Downloads\daemon.py", line 5, in <module>
    from dispatcher import TaskDispatcher
ModuleNotFoundError: No module named 'dispatcher'

How do I install daemon.py?
I can't handle Python.
Please let me know the answer.

Thanks.

**Comment by chopper-go** [comment]
You move daemon.py from project folder to your personal folder, I think this is wrong. And, I suppose that you have not do this "pip install -r requirements.txt"

**Comment by lily980905** [comment]
C:\python\webvideo-downloader>pip install -r requirements.txt
Requirement already satisfied: requests in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 1)) (2.24.0)
Requirement already satisfied: colorama in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 2)) (0.4.3)
Requirement already satisfied: chardet<4,>=3.0.2 in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (3.0.4)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (1.25.10)
Requirement already satisfied: idna<3,>=2.5 in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (2.10)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (2020.6.20)


I installed the requirements.txt file.
Setup failed to install daemon.py file only.
😭

**Comment by xyjz118** [comment]
> C:\python\webvideo-downloader>pip install -r requirements.txt
> Requirement already satisfied: requests in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 1)) (2.24.0)
> Requirement already satisfied: colorama in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from -r requirements.txt (line 2)) (0.4.3)
> Requirement already satisfied: chardet<4,>=3.0.2 in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (3.0.4)
> Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (1.25.10)
> Requirement already satisfied: idna<3,>=2.5 in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (2.10)
> Requirement already satisfied: certifi>=2017.4.17 in c:\users\lilyg\appdata\local\programs\python\python39\lib\site-packages (from requests->-r requirements.txt (line 1)) (2020.6.20)
> 
> I installed the requirements.txt file.
> Setup failed to install daemon.py file only.
> 😭

Check your scripts folder to see if you have been installed pip.
If not, You cannot run daemon.py without installing pip.

**Comment by lily980905** [comment]
The pip is installed in the Scripts folder.

![image](https://user-images.githubusercontent.com/72457510/95648430-feac8200-0b11-11eb-8594-4b72d32dee45.png)

**Comment by jaysonlong** [comment]
Assume that you have downloaded or cloned the project, and no file is moved, then just go to the directory **`webvideo-downloader/downloader`** and run:
```
python daemon.py
```
To install dependencies:
```
pip install -r requirements.txt
```
This project can not be installed now, you should cd the directory **`webvideo-downloader/downloader`** first before each run.

**Comment by lily980905** [comment]
Finally I succeeded in installing daemon.py file.
But I can't download the video.
The download button is missing.
Please give me answer.

I'm sorry to bother you.

Thanks.
![image](https://user-images.githubusercontent.com/72457510/95753043-f30cb700-0cdb-11eb-8f05-ca2002bd34ae.png)
![image](https://user-images.githubusercontent.com/72457510/95753120-16cffd00-0cdc-11eb-968b-52d7ac885232.png)

**Comment by jaysonlong** [comment]
Please describe your problem in detail if you want others to solve.
When and where does your problem occur? Show me the specific url or the example url.

**Comment by lily980905** [comment]
The hls downloader failed to read the file address.

daemon.py send me message : 
![image](https://user-images.githubusercontent.com/72457510/95758929-03289480-0ce4-11eb-95a8-3e032cc7ebcb.png)

**Comment by jaysonlong** [comment]
1. It seems that the download btn is not missing, that the violentmonkey script has no problem.
2. The output of the second pic means that your download has finished, you can find the video in the directory **`videos`** in the project folder.
3. The problem of the first pic is an encoding bug in my python scripts, I have already pushed the fixed code, just update the python script and it will works.

**Comment by lily980905** [comment]
I found the video folder.
Thank you.

But, when I updated the dispatcher.py file, I had this problem.


C:\webvideo-downloader\downloader>python daemon.py
Traceback (most recent call last):
  File "C:\webvideo-downloader\downloader\daemon.py", line 5, in <module>
    from dispatcher import TaskDispatcher
  File "C:\webvideo-downloader\downloader\dispatcher.py", line 6
    <!DOCTYPE html>
    ^
SyntaxError: invalid syntax
